app/src/main/java/com/example/githubclone/presentation/qrscanner/QRCodeAnalyzer.kt

```diff
@@ -0,0 +1,27 @@
+package com.example.githubclone.presentation.qrscanner
+
+import android.annotation.SuppressLint
+import androidx.camera.core.ImageProxy
+import com.google.mlkit.vision.barcode.BarcodeScanner
+import com.google.mlkit.vision.barcode.common.Barcode
+import com.google.mlkit.vision.common.InputImage
+import kotlinx.coroutines.tasks.await
+import timber.log.Timber
+
+@SuppressLint("UnsafeOptInUsageError")
+suspend fun processImageProxy(
+    barcodeScanner: BarcodeScanner,
+    imageProxy: ImageProxy,
+    onSuccess: (barcodes: List<Barcode>) -> Unit,
+) {
+    val inputImage =
+        InputImage.fromMediaImage(imageProxy.image!!, imageProxy.imageInfo.rotationDegrees)
```

Comment:
Let's try to never use the double bang `(!!)` operator unless absolutely required. Always check for nullability in Kotlin if possible.

Reply:
okay, I will make a note of it